core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -169,16 +176,21 @@ public FileIO io() {
     return io;
   }
 
-  private TableMetadata updateCurrentMetadata(LoadTableResponse response) {
-    // LoadTableResponse is used to deserialize the response, but config is not allowed by the REST
-    // spec so it can be
-    // safely ignored. there is no requirement to update config on refresh or commit.
-    if (current == null
-        || !Objects.equals(current.metadataFileLocation(), response.metadataLocation())) {
-      this.current = response.tableMetadata();
+  private void updateCurrentMetadata(LoadTableResponse response) {
+    this.current = response.tableMetadata();
+  }
+
+  private void updateIo(LoadTableResponse response) {
+    Map<String, String> mergedConfig = RESTUtil.merge(this.io.properties(), response.config());
+    String ioImpl = mergedConfig.getOrDefault(CatalogProperties.FILE_IO_IMPL, CatalogProperties.DEFAULT_FILE_IO_IMPL);
+    Object hadoopConf = null;
+    if (this.io instanceof HadoopConfigurable) {
+      hadoopConf = ((HadoopConfigurable) this.io).getConf();
     }
 
-    return current;
+    FileIO oldIo = this.io;
+    this.io = CatalogUtil.loadFileIO(ioImpl, mergedConfig, hadoopConf);
+    oldIo.close();
```

Comment:
Should we actually close the old IO? In [PyIceberg](https://github.com/apache/iceberg-python/blob/620ad9f64307193ec0d26846b48f4e063b5da904/pyiceberg/table/__init__.py#L811) we don't close it